src/BenchmarkDotNet/Running/BenchmarkRunnerDirty.cs

```diff
@@ -143,5 +144,66 @@ private static Summary[] RunWithExceptionHandling(Func<Summary[]> run)
                 return new[] { Summary.NothingToRun(e.Message, string.Empty, string.Empty) };
             }
         }
+
+        private static BenchmarkRunInfo[] TypeToBenchmarks(Type type, IConfig config)
+        {
+            if (type is null)
+                throw new InvalidBenchmarkDeclarationException("Type not provided.");
```

Comment:
For nulls we should throw `ArgumentNullException`

```suggestion
                throw new ArgumentNullException(paramName: nameof(type));
```

Reply:
Done (for arguments only, not for null values of the array) if I understood you correctly